apps/server/src/toasts/toasts.service.ts

```diff
@@ -66,4 +79,32 @@ export class ToastsService {
       await toast.destroy();
     }
   }
+  async countHappenedToastsInLastSixMonths(): Promise<number> {
+    const currentDate = new Date();
+    const { startDate } = this.getCurrentPeriod(currentDate);
+    return await this.toastModel.count({
+      where: {
+        date: {
+          [Op.gte]: startDate,
+          [Op.lte]: new Date(
+            startDate.getFullYear(),
+            startDate.getMonth() + 6,
+            1
+          ),
+        },
+        hasHappened: true,
+      },
+    });
+  }
+  async getToastsByUser(userId: string, hasHappened?: boolean): Promise<Toast[]> {
+    if(hasHappened!==undefined)
```

Comment:
I believe you can do `if(hasHappened)` instead of checking if it's not undefined.
nevertheless you need to add spaces to the if statement